client/my-sites/stats/pages/purchase/index.tsx

```diff
@@ -187,21 +192,44 @@ const StatsPurchasePage = ( {
 	return (
 		<Main fullWidthLayout>
 			<DocumentHead title={ translate( 'Jetpack Stats' ) } />
-			{ ! isLoading && (
-				<PageViewTracker
-					path="/stats/purchase/:site"
-					title="Stats > Purchase"
-					from={ query.from ?? '' }
-					variant={ variant }
-					is_upgrade={ +supportCommercialUse }
-					is_site_commercial={ isCommercial === null ? '' : +isCommercial }
-				/>
-			) }
+
 			<div
 				className={ classNames( 'stats', 'stats-purchase-page', {
 					'stats-purchase-page--is-wpcom': isTypeDetectionEnabled && isWPCOMSite,
 				} ) }
 			>
+				<NavigationHeader
+					className="stats__section-header modernized-header"
+					title={ translate( 'Jetpack Stats' ) }
+					subtitle={ translate(
+						"Gain insights into the activity and behavior of your site's visitors. {{learnMoreLink}}Learn more{{/learnMoreLink}}",
+						{
+							components: {
+								learnMoreLink: <InlineSupportLink supportContext="stats" showIcon={ false } />,
+							},
+						}
+					) }
+					navigationItems={ [] }
+				></NavigationHeader>
+				<StatsNavigation
+					selectedItem="traffic"
+					interval="day"
+					siteId={ siteId }
+					slug={ siteSlug }
+					showLock={ ! hasAnyPlan && ! isWPCOMSite && ! isVip }
```

Comment:
```suggestion
					showLock={ ! hasAnyPlan && isSiteJetpackNotAtomic && shouldShowPaywall }
```

Is it possible to align the lock display with the `redirectToPurchase` [check](https://github.com/Automattic/wp-calypso/blob/8205b42733fda9f754c3a69ee86a63ba5196bc16/client/my-sites/stats/stats-redirect/index.tsx#L58-L63) of the compulsory Stats redirect? I would get a little confused with the lock icons when my site without a subscription does not exceed the `MIN_MONTHLY_VIEWS_TO_APPLY_PAYWALL` since I can still access the Traffic and Insights pages.

https://github.com/Automattic/wp-calypso/assets/6869813/23e6be9a-7e3a-4b07-b6e2-92bbabc09143